### PR TITLE
catalog model tab shows only processors now

### LINKED ISSUE #292: Consider filter `model` category in `Catalog` page to be `processor`
### Please describe the feature that you want to propose

Currently the `model` category in `Catalog` page shows all camel models, including non-step (not a ProcessorDefinition) models like dataformats, languages, config objects, etc.
Screenshot from 2023-11-03 08-35-35

As a user, I'd like to see a list of processors instead, that could be used as a step. Even the contextual sub nodes like `GET`, `POST`, `PUT` and `DELETE` are eyesore to be mixed with processors.
We might want to see `dataformat` and `language` as a catalog, but they're better in a dedicated category, apart from processors.

For the catalog used for when user is adding a step, this is mandatory.

### PR DESCRIPTION
**Pull request by lhein**
fixes #292

### CODE REVIEW

**Comment by lhein** [comment]
grafik

**Comment by lhein** [comment]
There are still some entries that doesn't seem to belong there.

**Comment by lhein** [comment]
I added a blacklist for EIPs we don't want to have in the catalog.  These result looks much better now.

grafik

File: packages/camel-catalog/kaoto-camel-catalog-maven-plugin/src/main/java/io/kaoto/camelcatalog/CamelCatalogProcessor.java [hunk 1]
```diff
@@ -165,4 +169,21 @@ public String getPatternCatalog() throws Exception {
         });
         return JsonMapper.serialize(answer);
     }
+
+    private void populatePatterBlacklist() {
+        this.patternBlacklist.add("doCatch");
+        this.patternBlacklist.add("doFinally");
```
Comment (lordrip) [inline packages/camel-catalog/kaoto-camel-catalog-maven-plugin/src/main/java/io/kaoto/camelcatalog/CamelCatalogProcessor.java:175]: Maybe these 2 need to be commented out for the time being.

image

File: packages/camel-catalog/kaoto-camel-catalog-maven-plugin/src/main/java/io/kaoto/camelcatalog/CamelCatalogProcessor.java [hunk 2]
```diff
@@ -36,11 +37,14 @@ public class CamelCatalogProcessor {
     private final ObjectMapper jsonMapper;
     private final DefaultCamelCatalog api;
     private final CamelYamlDslSchemaProcessor schemaProcessor;
+    private final ArrayList<String> patternBlacklist;
```
Comment (lordrip) [inline packages/camel-catalog/kaoto-camel-catalog-maven-plugin/src/main/java/io/kaoto/camelcatalog/CamelCatalogProcessor.java:40]:
```suggestion
    private final ArrayList<String> patternBlocklist;
```

File: packages/camel-catalog/kaoto-camel-catalog-maven-plugin/src/main/java/io/kaoto/camelcatalog/CamelCatalogProcessor.java [hunk 3]
```diff
@@ -165,4 +169,21 @@ public String getPatternCatalog() throws Exception {
         });
         return JsonMapper.serialize(answer);
     }
+
+    private void populatePatterBlacklist() {
```
Comment (lordrip) [inline packages/camel-catalog/kaoto-camel-catalog-maven-plugin/src/main/java/io/kaoto/camelcatalog/CamelCatalogProcessor.java:173]:
```suggestion
    private void populatePatterBlocklist() {
```

**Review by lordrip — CHANGES_REQUESTED** [review]
Looks good, just a few pending :nail_care:

File: packages/camel-catalog/kaoto-camel-catalog-maven-plugin/src/main/java/io/kaoto/camelcatalog/CamelCatalogProcessor.java (hunk 2 above)
Comment (lhein) [reply packages/camel-catalog/kaoto-camel-catalog-maven-plugin/src/main/java/io/kaoto/camelcatalog/CamelCatalogProcessor.java]: Thank you for bringing this up. I fixed that.

File: packages/camel-catalog/kaoto-camel-catalog-maven-plugin/src/main/java/io/kaoto/camelcatalog/CamelCatalogProcessor.java (hunk 3 above)
Comment (lhein) [reply packages/camel-catalog/kaoto-camel-catalog-maven-plugin/src/main/java/io/kaoto/camelcatalog/CamelCatalogProcessor.java]: Thank you for bringing this up. I fixed that.

**Comment by lordrip** [comment]
I'll merge the pull request and work in having both catalogs

**Review by lordrip — APPROVED** [review]

**Pull request merged**